superset-frontend/src/components/FaveStar/FaveStar.test.tsx

```diff
@@ -0,0 +1,105 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import userEvent from '@testing-library/user-event';
+import FaveStar from '.';
+
+jest.mock('../Icon', () => ({
+  __esModule: true,
+  default: ({ name }: { name: string }) => (
+    <div data-test="icon" data-name={name} />
+  ),
+}));
+
+jest.mock('src/common/components/Tooltip', () => ({
+  Tooltip: (props: any) => <div data-test="tooltip" {...props} />,
+}));
+
+test('render right content', () => {
+  const props = {
+    itemId: 3,
+    saveFaveStar: jest.fn(),
+  };
+
+  const { rerender } = render(<FaveStar {...props} isStarred />);
+  expect(screen.getByRole('button')).toBeInTheDocument();
+  expect(screen.getByTestId('icon')).toBeInTheDocument();
+  expect(screen.getByTestId('icon')).toHaveAttribute(
+    'data-name',
+    'favorite-selected',
+  );
+
+  expect(props.saveFaveStar).toBeCalledTimes(0);
+  userEvent.click(screen.getByRole('button'));
+  expect(props.saveFaveStar).toBeCalledTimes(1);
+  expect(props.saveFaveStar).toBeCalledWith(props.itemId, true);
+
+  rerender(<FaveStar {...props} />);
+  expect(screen.getByTestId('icon')).toBeInTheDocument();
+  expect(screen.getByTestId('icon')).toHaveAttribute(
+    'data-name',
+    'favorite-unselected',
+  );
+
+  expect(props.saveFaveStar).toBeCalledTimes(1);
+  userEvent.click(screen.getByRole('button'));
+  expect(props.saveFaveStar).toBeCalledTimes(2);
+  expect(props.saveFaveStar).toBeCalledWith(props.itemId, false);
+});
+
+test('render content on tooltip', () => {
+  const props = {
+    itemId: 3,
+    showTooltip: true,
+    saveFaveStar: jest.fn(),
+  };
+
+  render(<FaveStar {...props} />);
+
+  expect(screen.getByTestId('tooltip')).toBeInTheDocument();
+  expect(screen.getByTestId('tooltip')).toHaveAttribute(
+    'id',
+    'fave-unfave-tooltip',
+  );
+  expect(screen.getByTestId('tooltip')).toHaveAttribute(
+    'title',
+    'Click to favorite/unfavorite',
+  );
+  expect(screen.getByRole('button')).toBeInTheDocument();
+});
+
+test('Call fetchFaveStar only on the first render', () => {
+  const props = {
+    itemId: 3,
+    fetchFaveStar: jest.fn(),
+    saveFaveStar: jest.fn(),
+    isStarred: false,
+    showTooltip: false,
+  };
+
+  const { rerender } = render(<FaveStar {...props} />);
+  expect(props.fetchFaveStar).toBeCalledTimes(1);
+  expect(props.fetchFaveStar).toBeCalledWith(props.itemId);
+
+  rerender(<FaveStar {...{ ...props, itemId: 2 }} />);
+  expect(props.fetchFaveStar).toBeCalledTimes(1);
+  // expect(screen.getByText('backend-name')).toBeInTheDocument();
```

Comment:
Seems like you forgot to remove a comment

Reply:
Tnks!